reliability/single-fit/rum/src/test/kotlin/com/datadog/android/rum/integration/ManualTrackingRumTest.kt

```diff
@@ -0,0 +1,407 @@
+            .hasRumEvent(0) {
+                hasService(stubSdkCore.getDatadogContext().service)
+                hasApplicationId(fakeApplicationId)
+                hasSessionType("user")
+                hasSource("android")
+                hasType("view")
+                hasViewUrl(viewKey)
+                hasViewName(viewName)
+                hasViewIsActive(true)
+                hasDocumentVersion(1)
+            }
+            .hasRumEvent(1) {
+                hasService(stubSdkCore.getDatadogContext().service)
+                hasApplicationId(fakeApplicationId)
+                hasSessionType("user")
+                hasSource("android")
+                hasType("view")
+                hasViewUrl(viewKey)
+                hasViewName(viewName)
+                hasViewIsActive(false)
+                hasDocumentVersion(2)
+            }
+    }
+
+    @RepeatedTest(16)
+    fun `M send view event with feature flag W startView() + addFeatureFlagEvaluation()`(
+        @StringForgery viewKey: String,
+        @StringForgery viewName: String,
+        @StringForgery ffKey: String,
+        @StringForgery ffValue: String
+    ) {
+        // Given
+        val rumMonitor = GlobalRumMonitor.get(stubSdkCore)
+
+        // When
+        rumMonitor.startView(viewKey, viewName, emptyMap())
+        rumMonitor.addFeatureFlagEvaluation(ffKey, ffValue)
+
+        // Then
+        val eventsWritten = stubSdkCore.eventsWritten(Feature.RUM_FEATURE_NAME)
+        print(eventsWritten[1].eventData)
+        StubEventsAssert.assertThat(eventsWritten)
+            .hasSize(2)
+            .hasRumEvent(0) {
+                hasService(stubSdkCore.getDatadogContext().service)
+                hasApplicationId(fakeApplicationId)
+                hasSessionType("user")
+                hasSource("android")
+                hasType("view")
+                hasViewUrl(viewKey)
+                hasViewName(viewName)
+                doesNotHaveField("feature_flag")
+                hasDocumentVersion(1)
+            }
+            .hasRumEvent(1) {
+                hasService(stubSdkCore.getDatadogContext().service)
+                hasApplicationId(fakeApplicationId)
+                hasSessionType("user")
+                hasSource("android")
+                hasType("view")
+                hasViewUrl(viewKey)
+                hasViewName(viewName)
+                hasFeatureFlag(ffKey, ffValue)
+                hasDocumentVersion(2)
+            }
+    }
+
+    @RepeatedTest(16)
+    fun `M send view event with action W startView() + addAction()`(
+        @StringForgery viewKey: String,
+        @StringForgery viewName: String,
+        @StringForgery actionName: String
+    ) {
+        // Given
+        val rumMonitor = GlobalRumMonitor.get(stubSdkCore)
+
+        // When
+        rumMonitor.startView(viewKey, viewName, emptyMap())
+        rumMonitor.addAction(RumActionType.CUSTOM, actionName, emptyMap())
+        Thread.sleep(100)
```

Comment:
custom action should be written immediately. Is this delay for the back propagation?

Reply:
It's only the case when another action is already active, otherwise it behaves like a normal action